data/devices/leeco.yml

```diff
@@ -114,3 +114,79 @@
   crypto:
     supported: true
     header_path: footer
+
+- name: LeEco Le 1S
+  id: x3
+  codenames:
+    - X3_HK
+    - X3
+    - x3
+    - x500
+    - X500
+  architecture: arm64-v8a
+ 
+  block_devs:
+    base_dirs:
+      - /dev/block/platform/mtk-msdc.0/by-name
+      - /dev/block/platform/mtk-msdc.0/11230000.MSDC0/by-name
+    system:
+      - /dev/block/platform/mtk-msdc.0/by-name/system
+      - /dev/block/platform/mtk-msdc.0/11230000.MSDC0/by-name/system
+      - /dev/block/mmcblk0p6
+    cache:
+      - /dev/block/platform/mtk-msdc.0/by-name/cache
+      - /dev/block/platform/mtk-msdc.0/11230000.MSDC0/by-name/cache
+      - /dev/block/mmcblk0p2
+    data:
+      - /dev/block/platform/mtk-msdc.0/by-name/userdata
+      - /dev/block/platform/mtk-msdc.0/11230000.MSDC0/by-name/userdata
+      - /dev/block/mmcblk0p3
+    boot:
+      - /dev/block/platform/mtk-msdc.0/by-name/boot
+      - /dev/block/platform/mtk-msdc.0/11230000.MSDC0/by-name/boot
+      - boot
+    recovery:
+      - /dev/block/platform/mtk-msdc.0/by-name/recovery
+      - /dev/block/platform/mtk-msdc.0/11230000.MSDC0/by-name/recovery
+      - recovery
+    extra:
+      - /dev/block/platform/mtk-msdc.0/by-name/metadata
+      - /dev/block/platform/mtk-msdc.0/11230000.MSDC0/by-name/metadata
+      - /dev/block/platform/mtk-msdc.0/by-name/nvram
+      - /dev/block/platform/mtk-msdc.0/11230000.MSDC0/by-name/nvram
+      - /dev/block/platform/mtk-msdc.0/by-name/lk
+      - /dev/block/platform/mtk-msdc.0/11230000.MSDC0/by-name/lk
+      - /dev/block/platform/mtk-msdc.0/by-name/logo
+      - /dev/block/platform/mtk-msdc.0/11230000.MSDC0/by-name/logo
+      - /dev/block/platform/mtk-msdc.0/by-name/para
+      - /dev/block/platform/mtk-msdc.0/11230000.MSDC0/by-name/para
+      - /dev/block/platform/mtk-msdc.0/by-name/tee1
+      - /dev/block/platform/mtk-msdc.0/11230000.MSDC0/by-name/tee1
+      - /dev/block/platform/mtk-msdc.0/by-name/tee2
+      - /dev/block/platform/mtk-msdc.0/11230000.MSDC0/by-name/tee2
+      - /dev/block/platform/mtk-msdc.0/by-name/secro
+      - /dev/block/platform/mtk-msdc.0/11230000.MSDC0/by-name/secro
+      - /dev/block/sda
+      - /dev/block/sda1
+      - /dev/block/mmcblk0p4
+      - misc
+ 
+  boot_ui:
+    supported: true
+    flags:
+      - TW_GRAPHICS_FORCE_USE_LINELENGTH
+      - TW_NO_USB_STORAGE
+      - TW_INCLUDE_CRYPTO
+      - TW_USE_MODEL_HARDWARE_ID_FOR_DEVICE_ID
+      - BOARD_SUPPRESS_SECURE_ERASE
```

Comment:
`TW_NO_USB_STORAGE`, `TW_INCLUDE_CRYPTO`, `TW_USE_MODEL_HARDWARE_ID_FOR_DEVICE_ID`, and `BOARD_SUPPRESS_SECURE_ERASE` are not needed by the boot UI and should be removed.